include/rocksdb/file_system.h

```diff
@@ -132,6 +132,14 @@ struct IODebugContext {
   // To be set by the FileSystem implementation
   std::string msg;
 
+  // To be set by the underlying FileSystem implementation.
+  std::string request_id;
+
+  // To be set by Warm Storage if underlying filesystem is warm storage.
+  // This bool is used during IO tracing to add additional information in trace
+  // records.
+  bool is_ws = 0;
```

Comment:
Warm Storage is an FB intetrnal component. We don't want to pollute RocksDB code with references to such components, and even any specific file system for that matter. Is this to decide whether to add ```request_id``` to the trace? Perhaps we can call this flag ```fs_info``` or something generic, and provide a ```SetRequestID()``` function that will hide these details.

Reply:
@anand1976 I was wondering should I add the logic I added in first PR and hide the implementation in SetRequestID? That way underlying FileSystem had to call SetRequestId only and i can add the logic of whether to trace that data inside that function.

Reply:
I think the bitwise logic is fine, since we're hiding it anyway.